labelbox/data/serialization/coco/panoptic_dataset.py

```diff
@@ -0,0 +1,178 @@
+
+
+def mask_to_coco_segment_info(canvas: np.ndarray, annotation,
+                              annotation_idx: int, category_id):
+    color = id_to_rgb(annotation_idx)
+    mask = annotation.value.draw(color=color)
+    shapely = annotation.value.shapely
+    xmin, ymin, xmax, ymax = shapely.bounds
+    canvas = np.where(canvas == (0, 0, 0), mask, canvas)
+    return SegmentInfo(id=annotation_idx,
+                       category_id=category_id,
+                       area=shapely.area,
+                       bbox=[xmin, ymin, xmax - xmin, ymax - ymin]), canvas
+
+
+def process_label(label: Label,
+                  idx: Union[int, str],
+                  image_root,
+                  mask_root,
+                  all_stuff=False):
+    """
+    Masks become stuff
+    Polygon and rectangle become thing
+    """
+    annotations = get_annotation_lookup(label.annotations)
+    image_id = get_image_id(label, idx)
+    image = get_image(label, image_root, image_id)
+    canvas = np.zeros((image.height, image.width, 3))
+
+    segments = []
+    categories = {}
+    is_thing = {}
+
+    for class_idx, class_name in enumerate(annotations):
+        for annotation_idx, annotation in enumerate(annotations[class_name]):
+            categories[annotation.name] = hash_category_name(annotation.name)
+            if isinstance(annotation.value, Mask):
+                segment, canvas = (mask_to_coco_segment_info(
+                    canvas, annotation, class_idx + 1,
+                    categories[annotation.name]))
+                segments.append(segment)
+                is_thing[annotation.name] = 0
+
+            elif isinstance(annotation.value, (Polygon, Rectangle)):
+                segment, canvas = vector_to_coco_segment_info(
+                    canvas,
+                    annotation,
+                    annotation_idx=(class_idx if all_stuff else annotation_idx)
+                    + 1,
+                    image=image,
+                    category_id=categories[annotation.name])
+                segments.append(segment)
+                is_thing[annotation.name] = 1 - int(all_stuff)
+
+    mask_file = image.file_name.replace('.jpg', '.png')
+    mask_file = os.path.join(mask_root, mask_file)
+    Image.fromarray(canvas.astype(np.uint8)).save(mask_file)
+    return image, PanopticAnnotation(
+        image_id=image_id,
+        file_name=mask_file.split(os.sep)[-1],
+        segments_info=segments), categories, is_thing
+
+
+class CocoPanopticDataset(BaseModel):
+    info: Dict[str, Any] = {}
+    images: List[CocoImage]
+    annotations: List[PanopticAnnotation]
+    categories: List[Categories]
+
+    @classmethod
+    def from_common(cls, labels: LabelCollection, image_root, mask_root,
+                    all_stuff):
+        all_coco_annotations = []
+        coco_categories = {}
+        coco_things = {}
+        images = []
+        with ProcessPoolExecutor(max_workers=8) as exc:
```

Comment:
Same comment about executor as arg. Not being able to specify concurrency would be pretty annoying to me as user

Reply:
What if I just let the number of workers be an arg? My thinking is that most users are going to want the same amount of parallelism - the max that the backend can handle. Also this script is unlikely to run within the context of another executor - users are going to either convert an entire dataset or not. So asking users to pass in the execturor is an extra step that will always be necessary.